### PR TITLE
Feature: Add loading state to RestartDialog

### PR DESCRIPTION
**Pull request by ThatGalNatalie**
PR Checklist:

- [ ] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Adds a loading and disabled state when restarting flow runs

### CODE REVIEW

**Comment by stackoverfloweth** [comment]
Screen Shot 2021-11-30 at 9 40 10 AM

File: src/pages/FlowRun/Actions.vue [hunk 1]
```diff
@@ -183,7 +183,9 @@ export default {
             depressed
             style="height: 46px;"
             small
-            :disabled="!hasPermission('update', 'run') || !canRestart"
+            :disabled="
+              !hasPermission('update', 'run') || !canRestart || restartDialog
```
Comment (stackoverfloweth) [inline src/pages/FlowRun/Actions.vue:187]: is this change necessary? that dialog is a modal, so the user shouldn't be able to click anything behind the modal when `restartDialog` is true

Comment (stackoverfloweth) [reply src/pages/FlowRun/Actions.vue:187]: although, visually I do like the effect

**Review by stackoverfloweth — APPROVED** [review]
LGTM

Great addition!
It would be nice if when the restart-dialog closes, the flow-run behind should appear in a "scheduled" or "running" state. Immediately after closing restarting, the flow-run behind still appears cancelled, and I can go through the restart dialog again. I'm not suggesting this should be done as part of your PR.

**Comment by zhen0** [comment]
Added https://github.com/PrefectHQ/ui/issues/1147 to capture Evan's suggestion here.
